app/scripts/modules/core/src/confirmationModal/ConfirmModal.tsx

```diff
@@ -0,0 +1,133 @@
+import { IModalComponentProps, Markdown } from 'core/presentation';
+import { NgReact } from 'core/reactShims';
+import { MultiTaskMonitor } from 'core/task/monitor/MultiTaskMonitor';
+import { ModalClose } from 'core/modal';
+import { PlatformHealthOverride } from 'core/application/modal/PlatformHealthOverride';
+import { IConfirmationModalPassthroughProps } from 'core';
+
+export interface IConfirmModalProps extends IModalComponentProps, IConfirmationModalPassthroughProps {
+  taskMonitor?: TaskMonitor;
+  taskMonitors?: TaskMonitor[];
+}
+
+export const ConfirmModal = (props: IConfirmModalProps) => {
+  const { account, verificationLabel, textToVerify, taskMonitor, taskMonitors, closeModal, dismissModal } = props;
+  const { useState, useEffect } = React;
+
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [isRetry, setIsRetry] = useState(false);
+  const [isValid, setIsValid] = useState(!account && !verificationLabel && !textToVerify);
+  const [error, setError] = useState<{ isError: boolean; message?: string }>({ isError: false });
+
+  const [reason, setReason] = useState<string>();
+  const [interestingHealthProviderNames, setInterestingHealthProviderNames] = useState<string[]>();
+
+  useEffect(() => {
+    if (taskMonitor && !taskMonitor.modalInstance) {
+      taskMonitor.modalInstance = TaskMonitor.modalInstanceEmulation(closeModal, dismissModal);
+    }
+    if (taskMonitor?.onTaskRetry) {
+      const { onTaskRetry } = taskMonitor;
+      taskMonitor.onTaskRetry = () => {
+        setIsRetry(true);
+        setIsSubmitting(false);
+        onTaskRetry();
+      };
+    }
+  }, [taskMonitor]);
+
+  const requiresVerification = account || (verificationLabel && textToVerify);
+  const isDisabled = isSubmitting || !isValid;
+
+  const showError = (e: string) => {
+    setError({ isError: true, message: e });
+    setIsSubmitting(false);
+  };
+
+  const submit = () => {
+    const { submitMethod, submitJustWithReason } = props;
+    if (isDisabled) {
+      return;
+    }
+    setIsSubmitting(true);
+    if (taskMonitors) {
+      taskMonitors.forEach(m => m.callPreconfiguredSubmit({ reason }));
+    } else if (taskMonitor) {
+      taskMonitor.submit(() => {
+        return submitMethod({
+          interestingHealthProviderNames,
+          reason,
+        });
+      });
+    } else if (submitJustWithReason) {
+      submitMethod({ reason }).then(dismissModal);
+    } else {
+      if (submitMethod) {
+        submitMethod().then(dismissModal, showError);
+      } else {
+        closeModal();
+      }
+    }
+  };
+
+  const { TaskMonitorWrapper, ButtonBusyIndicator } = NgReact;
+  const includeBody =
+    (isRetry && props.retryBody) ||
+    props.bodyContent ||
+    error.isError ||
+    props.platformHealthOnlyShowOverride ||
+    props.askForReason ||
+    props.submitJustWithReason;
+
+  return (
+    <div>
+      <TaskMonitorWrapper monitor={taskMonitor} />
```

Comment:
can we use the `<Modal.Header/>`, `<Modal.Body/>` and `<Modal.Footer/>` components?  or maybe @erikmunson 's modal rewrite?

Reply:
`<Modal.Header/>`, `<Modal.Body/>`, `<Modal.Footer/>` all worked.

Since this isn't necessarily called from another component, using the modal rewrite would be a little clunky in this case (I'd have to do some node mounting business)